src/structures/PermissionOverwrites.js

```diff
@@ -164,24 +169,29 @@ class PermissionOverwrites {
   /**
    * Resolves an overwrite into {@link RawOverwriteData}.
    * @param {OverwriteResolvable} overwrite The overwrite-like data to resolve
-   * @param {Guild} guild The guild to resolve from
+   * @param {Guild} [guild] The guild to resolve from
    * @returns {RawOverwriteData}
    */
   static resolve(overwrite, guild) {
     if (overwrite instanceof this) return overwrite.toJSON();
-    if (typeof overwrite.id === 'string' && ['role', 'member'].includes(overwrite.type)) {
-      return { ...overwrite, allow: Permissions.resolve(overwrite.allow), deny: Permissions.resolve(overwrite.deny) };
+    if (typeof overwrite.id === 'string' && OverwriteTypes.includes(overwrite.type)) {
```

Comment:
This will throw, HARD. OverwriteTypes is an Object, not an array, it doesn't have `includes`. Use `Reflect.has(OverwriteTypes, overwrite.type)` or it's `in` variant (`overwrite.type in OverwriteTypes`